### ISSUE TITLE
How do you get at events in withStateHandlers?

### ISSUE THREAD

**Issue by GollyJer** [acdlite/recompose#456]
Here's a contrived example (I'm using React Native).
```
const App = ({ positionX, handleButtonPressX, positionY, handleButtonPressY }) =>
  (<View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
    <Button title="Set X Position" onPress={handleButtonPressX} />
    <Button title="Set Y Position" onPress={handleButtonPressY} />
    <Text style={{ marginTop: 30 }}>
      {`X: ${positionX}\nY: ${positionY}`}
    </Text>
  </View>);

const trackPositionX = compose(
  withState('positionX', 'setPositionX', ''),
  withHandlers({
    handleButtonPressX: ({ setPositionX }) => (event) => {
      setPositionX(Math.floor(event.nativeEvent.pageX));
    },
  }),
);

const trackPositionY = withStateHandlers(
  () => ({
    positionY: '',
  }),
  {
    handleButtonPressY: () => (event) => ({
      positionY: Math.floor(event.nativeEvent.pageY),
    }),
  },
);
```

When clicking the *Set Y Position* button I get the error  
 `null is not an object (evaluating 'event.nativeEvent.pageY')`

It seems  `event` is undefined. I know this is probably a basic programming concept I'm not getting. Can someone help me understand?

Thanks!

**Comment by istarkov** [comment]
At the moment of setState state function is called event can be already freed, as React reuses event object.
Seems like we need to fix this somehow.

**Comment by istarkov** [comment]
It will be a good first PR

**Comment by istarkov** [comment]
For more details plz read React docs about event.persist

**Comment by GollyJer** [comment]
I really want to help with this but am completely out of my element even getting things set up.
I've forked the repo and cloned locally but don't know what to do next.

How do I set up my local project to use the cloned version of recompose?  I feel that will be the easiest way for me to test if my changes are working.  Is there a better way?

Sorry, total newb at contributing to an external project.
